paper/paper.md

```diff
@@ -0,0 +1,173 @@
+
+Once the project structure and navigation has been established, there are several functions that help the scientific workflow. For example, the function `dict_list` provides a convient and consistent way of defining containers of parameter values. 
+`savename` can be used for preparing a file name or a figure title. 
+Using it would transform the following dictionary
+```julia
+parameters = Dict(:phi => 3, :pos_z => 0.5, :date => Date(2020,5,23))
+```
+into 
+```julia
+savename(parameters, "bson")
+```
+```
+"date=2020-05-23_pos_z=0.5_phi=3.bson"
+```
+
+Once a simulation script is created, taking advantage of e.g. `projectdir, dict_list, savename` among other functions, the user will typically want to save numeric results on disk.
+DrWatson offers many functions that help the workflow at this level.
+`safesave` ensures that saved data will never overwrite existing files, but make a new version instead (and back up the original file).
+`tagsave` allows one to add git-related information to saved data.
+`tagsave` is a function that excellently highlights how DrWatson is a minimally invasive framework.
+Continuing from the dictionary `parameters` defined above, we would save any kind of `data` wrapped in a dictionary with the command `save(savename(parameters, "bson"), data)` (ending `.bson` is used as an example).
+By only replacing the function `save` with `@tagsave`, it is possible to attach important information to the saved data
+
+```julia
+@tagsave(savename(parameters, "bson"), Dict("data" => [1,2,3]));
+
+load(savename(parameters, "bson")) # load back saved data
+```
+
+```
+Dict{Symbol,Any} with 6 entries:
+  :gitcommit => "v1.13.0-1-g3a5364f"
+  :script    => "docs/build/string#3"
+  :data      => [1, 2, 3]
+  :gitpatch  => ""
+```
+
+The fields `:gitcommit, :script, :gitpatch` were added automatically and provide the necessary information for reproducibility. In case of uncommited modifications (also called a "dirty git repository"), a patch is saved under the `:gitpatch` key which can be applied to the `:gitcommit` to restore the exact state of the repository.
+Calling `@tagsave` without extra arguments assumed that you used DrWatson's suggested folder structure using `initialize_project`, and thus could find all git-related information automatically.
+However this is not necessary, you can instead provide a keyword argument `gitpath` to `@tagsave` and explictly specify a git path.
+Finally, the `parse_savename(filename; kwargs...)` function can be used to obtain the parameters dictionary from the filename.
+
+A last step is data analysis.
+There the function `collect_results` is a powerful tool that can automatically collect all saved output files into a `DataFrame` (the major Julia tabular datastructure) for further analysis.
+The function is adaptive in the sense that adding new simulations (with potentially different parameter names) automatically adjusts existing columns to have `missing` values at appropriate locations.
```

Comment:
We were discussing whether we want to keep this section or remove it to shorten the workflow, @JonasIsensee ?

Reply:
That is correct.
Last time I worked on this I couldn't quite decide on how to shorten this.
I'll try again.

Reply:
@Datseris 
I rewrote that part of the workflow section to make it shorter and more concise.
If you like it, then we're done.
Otherwise you may also remove the paragraph completely. I'm fine with it either way.